docs(chart_template_guide): point YAML Anchors at library charts (#2117)
helm/helm-www: docs/chart_template_guide/yaml_techniques.md

Because Helm and Kubernetes often read, modify, and then rewrite YAML files, the
anchors will be lost.

:::info

> Reaching for anchors as a DRY mechanism? For sharing snippets across charts,

Reviewer: ```suggestion
For sharing snippets across charts,
```

> prefer [library charts](/topics/library_charts.md) — they are designed for

Reviewer: ```suggestion
use [library charts](/topics/library_charts.md) instead of YAML anchors.
```

> reuse and aren't affected by the round-trip pitfall described above.

Reviewer: ```suggestion
Library charts are designed for reuse,
and aren't subject to the round-trip pitfall described above.
```
